compiler/ml/untypeast.ml

```diff
@@ -261,7 +261,8 @@ let pattern sub pat =
       | Tpat_record (list, closed) ->
         Ppat_record
           ( List.map
-              (fun (lid, _, pat) -> (map_loc sub lid, sub.pat sub pat))
+              (fun (lid, _, pat, opt) ->
+                (map_loc sub lid, sub.pat sub pat, opt))
```

Comment:
This is where the extra field in typed ASTs is used: to compile back to untyped ASTs.
Before, this was passed down in attributes.

The nest phase, translation to lambda, does not need this information.